lib/waterline/query/aggregate.js

```diff
@@ -51,8 +53,49 @@ module.exports = {
       return value !== undefined;
     });
     
-    // Create will take care of cloning values so original isn't mutated
-    async.map(filteredValues, self.create.bind(self), cb);
+    // process each item, their associations and beforeCreate callbacks
```

Comment:
> The call to `self.adapter.createEach()` method in my implementation actually calls the underlined createEach() method inside adapter/aggregateQueries.js, which already handles if the adapter implements createEach or not.

This is why I believe the values are being processed twice when the adapter doesn't implement `.createEach()`. The values are being processed here once and then again inside [waterline's `.createEach()`](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/dql/create.js#L65-L76). This will make performance even worse for adapters not implementing `.createEach()`... I haven't yet tested the change but that's what it looks like....